pkg/util/provider/machinecontroller/machine_util.go

```diff
@@ -231,6 +231,38 @@ func mergeDataMaps(in map[string][]byte, maps ...map[string][]byte) map[string][
 	return out
 }
 
+// syncMachineNameToNode syncs the machine name on the corresponding node object
+// by adding a machine name label to its metadata.
+func (c *controller) syncMachineNameToNode(ctx context.Context, machine *v1alpha1.Machine) (machineutils.RetryPeriod, error) {
+	node, err := c.nodeLister.Get(getNodeName(machine))
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// Don't return error so that other steps can be executed.
+			return machineutils.LongRetry, nil
+		}
+		klog.Errorf("Error occurred while trying to fetch node object - err: %s", err)
+		return machineutils.ShortRetry, err
+	}
+
+	if node.Labels[machineutils.MachineLabelKey] == machine.Name {
+		return machineutils.LongRetry, nil
+	}
+
```

Comment:
can you make a DeepCopy of the node and then update it? I have seen the update call failing otherwise.